### PR TITLE
`ParticleLevelSet` accessors

### PR DESCRIPTION
**Pull request by abisner**
Added accessors for color indices and color count.

### CODE REVIEW

File: src/Picasso_ParticleLevelSet.hpp [hunk 1]
```diff
@@ -231,6 +231,12 @@ class ParticleLevelSet
         _radius = _dx * params.get<double>( "particle_radius", 0.5 );
     }
 
+    Kokkos::View<int*, memory_space> colorIndex() const
+    {
+        return _color_indices;
+    }
+    int colorCount() const { return _color_count; }
```
Comment (sslattery) [inline src/Picasso_ParticleLevelSet.hpp:238]: need a blank line between these definitions

File: src/Picasso_ParticleLevelSet.hpp [hunk 2]
```diff
@@ -231,6 +231,12 @@ class ParticleLevelSet
         _radius = _dx * params.get<double>( "particle_radius", 0.5 );
     }
 
+    Kokkos::View<int*, memory_space> colorIndex() const
```
Comment (dalg24) [inline src/Picasso_ParticleLevelSet.hpp:234]: Do you mean to return non-const view?

Comment (sslattery) [reply src/Picasso_ParticleLevelSet.hpp:234]: yes this should be `const int*`

**Review by streeve — APPROVED** [review]
Ready to merge if you are @abisner

**Comment by abisner** [comment]
The latest commit was needed to allow reuse of `ParticleLevelSetPrimitiveData` type outside of `ParticleLevelSet` class, along with the color index accessor, which now returns a `const int*` view. The `ParticleLevelSetPrimitiveData.c` data member is now a ViewType template, allowing copy assignment of a `const-` view.

**Review by streeve — APPROVED** [review]

**Pull request merged**